x/supplier/keeper/proof.go

```diff
@@ -9,52 +11,51 @@ import (
 
 // UpsertProof inserts or updates a specific proof in the store by index.
 func (k Keeper) UpsertProof(ctx sdk.Context, proof types.Proof) {
```

Comment:
Ignore/resolve if this is addressed / discussed in a different.

Since we have `Upsert` now, we need to make sure that `Create or Update` only happens if the proof is signed by the supplier who created. 

Do we have a `TODO_BLOCKER` for it?

Reply:
The TODO is in the supplier keeper's `#SubmitProof()` method (and the analogous claim todo is in `#CreateClaim()`).